src/app/static/src/app/components/plots/choropleth/Choropleth.vue

```diff
@@ -0,0 +1,135 @@
+const featureData = ref<IndicatorValuesDict>({});
+
+const map = ref<typeof LMap | null>(null);
+const featureRefs = ref<typeof LGeoJson[]>([]);
+
+const {createTooltips, updateTooltips} = useMapTooltips(featureData, indicatorMetadata, currentFeatures, featureRefs)
+
+const updateMap = () => {
+    updateFeatures();
+    updateColourScales();
+    updateTooltips();
+};
+// Watch on the plotData only instead of using computed. Here we want to update the colours,
+// the features and the legend. If we use computed this will cause multiple updates
+// 1. When a selection changes it will update immediately
+// 2. After the plotData has been fetched async it will update again
+// This can cause the map to appear to flicker as it updates with old scales
+// then quickly updates after new data has been fetched.
+// Instead manually watch on the plot data changes, and also trigger this when a user changes the
+// scale selection
+watch([plotData], updateMap)
+
+const updateColourScales = () => {
+    const colourScales = store.state.plotState.output.colourScales;
+    const selectedIndicator =  store.getters["plotSelections/selectedIndicator"];
+    selectedScale.value = colourScales[selectedIndicator];
+    if (!selectedScale.value) {
+        selectedScale.value = initialiseScaleFromMetadata(indicatorMetadata.value);
+        updateOutputColourScale(selectedScale.value);
+    }
+    colourRange.value = getColourRange(indicatorMetadata.value, selectedScale.value, plotData.value);
+    scaleLevels.value = getScaleLevels(indicatorMetadata.value, colourRange.value);
+    featureData.value = getFeatureData(
+            plotData.value,
+            indicatorMetadata.value,
+            colourRange.value ? colourRange.value : {max: 1, min: 0}
+    );
+}
+
+const updateFeatures = () => {
+    indicatorMetadata.value = store.getters["modelCalibrate/indicatorMetadata"];
+    const selectedLevel = store.state.plotSelections.choropleth.filters
+            .find(f => f.stateFilterId === "detail")!.selection;
+    const selectedAreas = store.state.plotSelections.choropleth.filters
+            .find(f => f.stateFilterId === "area")!.selection;
+    currentFeatures.value = getVisibleFeatures(features, selectedLevel, selectedAreas);
+}
+
+const updateBounds = () => {
+    if (featureRefs.value.length > 0) {
+        map.value?.leafletObject.fitBounds(featureRefs.value.map(f => f.leafletObject.getBounds()));
+    }
+};
+
+watch(featureRefs.value, () => {
+    updateBounds();
+});
+
+const showColour = (feature: Feature) => {
+    return featureData.value[feature.properties!.area_id]
+};
+
+const getColour = (feature: Feature) => {
+    if (showColour(feature)) {
+        return featureData.value[feature.properties!.area_id].color;
+    } else {
+        //show a lighter grey than the outlines if no data
+        //so unselected regions are still distinguishable
+        return "rgb(200,200,200)";
+    }
+};
+
+const style = {
+    className: "geojson"
+};
```

Comment:
minor comment, this doesnt really change, can we not just put this into the `options-style` instead of spreading it into every feature

or you can make a `getStyle` method that accepts feature and returns a callback that returns
```javascript
{
  className: "geojson",
  fillColor: getColor(feature)
}
```